src/core/annotations/qgsannotationitem.h

```diff
@@ -144,16 +145,30 @@ class CORE_EXPORT QgsAnnotationItem
     /**
      * Applies an edit \a operation to the item.
      *
-     * \since QGIS 3.22
+     * \deprecated QGIS 3.40, use applyEditV2() instead.
      */
-    virtual Qgis::AnnotationItemEditOperationResult applyEdit( QgsAbstractAnnotationItemEditOperation *operation );
+    Q_DECL_DEPRECATED virtual Qgis::AnnotationItemEditOperationResult applyEdit( QgsAbstractAnnotationItemEditOperation *operation ) SIP_DEPRECATED;
+
+    /**
+     * Applies an edit \a operation to the item.
+     *
+     * \since QGIS 3.40
+     */
+    virtual Qgis::AnnotationItemEditOperationResult applyEditV2( QgsAbstractAnnotationItemEditOperation *operation, const QgsAnnotationItemEditContext &context );
 
     /**
      * Retrieves the results of a transient (in progress) edit \a operation on the item.
      *
-     * \since QGIS 3.22
+     * \deprecated QGIS 3.40, use transientEditResultsV2() instead.
+     */
+    Q_DECL_DEPRECATED virtual QgsAnnotationItemEditOperationTransientResults *transientEditResults( QgsAbstractAnnotationItemEditOperation *operation ) SIP_FACTORY;
+
+    /**
+     * Retrieves the results of a transient (in progress) edit \a operation on the item.
+     *
+     * \since QGIS 3.40
      */
-    virtual QgsAnnotationItemEditOperationTransientResults *transientEditResults( QgsAbstractAnnotationItemEditOperation *operation ) SIP_FACTORY;
+    virtual QgsAnnotationItemEditOperationTransientResults *transientEditResultsV2( QgsAbstractAnnotationItemEditOperation *operation, const QgsAnnotationItemEditContext &context ) SIP_FACTORY;
```

Comment:
are you forced to create a V2? you cannot keep the same name (with just adding the extra argument)?

Reply:
Sip says no -- that was my first approach, but it makes it so neither method can be called from python 🤷‍♂️

Reply:
weird!
what about this:

```c++
    Q_DECL_DEPRECATED virtual QgsAnnotationItemEditOperationTransientResults *transientEditResults( QgsAbstractAnnotationItemEditOperation *operation, bool dummyVariable = True )  SIP_DEPRECATED SIP_FACTORY;

    virtual QgsAnnotationItemEditOperationTransientResults *transientEditResults( QgsAbstractAnnotationItemEditOperation *operation, const QgsAnnotationItemEditContext &context ) SIP_FACTORY;
```

That would avoid any renaming for 4.0.

Reply:
But that change in signature would break any python item classes implementing the method..?